phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/IndexUpgradeTool.java

```diff
@@ -367,6 +358,53 @@ private int executeTool(Connection conn, ConnectionQueryServices queryServices,
         return 0;
     }
 
+    private void executeToolForImmutableTables(ConnectionQueryServices queryServices,
```

Comment:
I suggest having a separate list, i.e., mutableList, for mutable tables.  We should disable and modify immutable tables first and then record the current time, say as startWaitTime. Then we can upgrade mutable tables (disable, modify and rebuildIndexes). After that we can  sleep for 11 minutes - (current time - startWaitTime) and call rebuildIndexes for mutable tables. This eliminates unnecessary sleep time better.